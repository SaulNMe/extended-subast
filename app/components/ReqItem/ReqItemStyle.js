import { StyleSheet } from 'react-native';
import { Fonts, Colors } from 'cotizame-native/app/styles';
import { ApplicationStyles } from 'cotizame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	dateStyle: {
		fontSize: Fonts.size.small,
		fontWeight: Fonts.weight.regular
	},
	imageStyle: {
		width: 60,
		height: 60,
		borderRadius:10
	},
	redText: {
		color: Colors.red
	}
});
