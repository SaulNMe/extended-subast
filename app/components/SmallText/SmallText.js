import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
	Text,
	View
} from 'react-native';

import styles from './SmallTextStyle';
import { Fonts, Colors } from 'cotizame-native/app/styles';

export default class SmallText extends Component {
	render() {
		const textColor = this.props.color ? { color: Colors[this.props.color] }: { color: Colors.black };
		const weight = this.props.weight ? { fontWeight: Fonts.weight[this.props.weight] }: { fontWeight: Fonts.weight.regular };
		const center = this.props.center ? { textAlign: 'center' } : {};

		return (
			<React.Fragment>
				<Text style={[styles.smallText, textColor, weight, center]}>
					{this.props.text}
				</Text>
			</React.Fragment>
		);
	}
}

SmallText.propTypes = {
	text: PropTypes.string
}

SmallText.defaultProps = {
	text: ""
}
