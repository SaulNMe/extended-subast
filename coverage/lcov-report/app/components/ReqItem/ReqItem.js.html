<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/ReqItem/ReqItem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">app/components/ReqItem</a> ReqItem.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
&nbsp;
import {
	Text,
	View,
	Image,
	TouchableOpacity
} from 'react-native';
&nbsp;
import styles from './ReqItemStyle';
import SmallText from 'cotizame-native/app/components/SmallText';
import BodyText from 'cotizame-native/app/components/BodyText';
import Feather from 'react-native-vector-icons/Feather';
import RNMomentCountDown from 'react-native-moment-countdown';
import { Colors } from 'cotizame-native/app/styles';
import moment from 'moment';
import Local from 'cotizame-native/app/services/Local';
&nbsp;
&nbsp;
export default class ReqItem extends Component {
	render(){ 
		let endTime  = this.props.endTime;
		let beginningTime  = String(moment(this.props.time).format('YYYY-MM-DD HH:mm:ss'));
		return (
			&lt;View&gt;
				&lt;View style={[styles.row, this.props.marginHorizontal &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.marginHorizontal]</span>}&gt;
					{!this.props.noImage &amp;&amp;
						&lt;View style={[styles.justifyContentCenter, styles.marginRight]}&gt;
							&lt;Image source={require('cotizame-native/assets/msgLogo.png')} style={{ width: 55, height: 55}}/&gt;
						&lt;/View&gt;
					}
					&lt;View style={styles.flex1}&gt;
						&lt;View style={[styles.row]}&gt;
							&lt;View style={styles.flex1}&gt;
								&lt;SmallText
									text= {this.props.label} 
									color='subdued'
								/&gt;
								&lt;BodyText
									text= {this.props.title}
									color='black'
									weight='medium'
								/&gt;
							&lt;/View&gt;
							{!this.props.noMoreButton &amp;&amp; 
								&lt;View&gt;
									&lt;TouchableOpacity&gt;
										&lt;Feather name='more-vertical' size={24} color={Colors.light}/&gt;
									&lt;/TouchableOpacity&gt;
								&lt;/View&gt;
							}
						&lt;/View&gt;
						&lt;View style={[styles.row, styles.justifyContentSpaceBetween]}&gt;
							&lt;View style={[styles.justifyContentSpaceBetween, styles.flex1, styles.row]}&gt;
								&lt;View style={styles.column}&gt;
									&lt;SmallText
										text={Local.get('reqItem.bestPrice')}
										color='subdued'
									/&gt;
									&lt;SmallText 
										text={'$' + this.props.price + ' MXN'}
										color='darker'
										weight='regular'
									/&gt;
								&lt;/View&gt;
								&lt;View style={styles.column}&gt;
									&lt;SmallText
										text={Local.get('reqItem.time')}  
										color='subdued'
									/&gt;
									{moment(beginningTime).isBefore(endTime) ?
<span class="branch-0 cbranch-no" title="branch not covered" >										&lt;SmallText</span>
											text='Expired'
											color='red'
										/&gt;
										:
										&lt;RNMomentCountDown
						 					timeStyle={styles.dateStyle}
						 					toDate= {this.props.time}
						 					sourceFormatMask='YYYY-MM-DD HH:mm:ss'
						 				 /&gt;
									}
								&lt;/View&gt;
							&lt;/View&gt;
						&lt;/View&gt;
					&lt;/View&gt;
				&lt;/View&gt;
			&lt;/View&gt;
		);}
}
&nbsp;
ReqItem.propTypes = {
	label: PropTypes.string,
	title: PropTypes.string,
	price: PropTypes.number,
	time: PropTypes.string,
	endTime: String(moment().format('YYYY-MM-DD HH:mm:ss')),
	marginHorizontal: PropTypes.bool,
	noImage: PropTypes.bool,
	noMoreButton: PropTypes.bool,
}
&nbsp;
ReqItem.defaultProps = {
	label: 'label',
	title: 'title',
	price: 0,
	time: 'time',
	endTime: String(moment().format('YYYY-MM-DD HH:mm:ss')),
	marginHorizontal: false,
	noImage: false,
	noMoreButton: false,
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:35:10 GMT-0600 (Central Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
