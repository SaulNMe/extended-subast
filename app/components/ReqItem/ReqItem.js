import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
	Text,
	View,
	Image,
	TouchableOpacity
} from 'react-native';

import styles from './ReqItemStyle';
import SmallText from 'cotizame-native/app/components/SmallText';
import BodyText from 'cotizame-native/app/components/BodyText';
import Feather from 'react-native-vector-icons/Feather';
import RNMomentCountDown from 'react-native-moment-countdown';
import { Colors } from 'cotizame-native/app/styles';
import moment from 'moment';
import Local from 'cotizame-native/app/services/Local';


export default class ReqItem extends Component {
	constructor (props) {
		super(props)
		this.state = { endCountdown: false, hourLimit: false }
	}

	countdownEnd = () => {
		this.setState({ endCountdown: true });
	}

	handleUrl = (url) => {
		let newUrl = {uri: 'https://drive.google.com/uc?id='+ url};
		return newUrl;
	}

	onHourLimit = (expiration) => {
		let hour = String(moment().add(1, 'h').format('YYYY-MM-DD HH:mm:ss'));
		return moment(expiration).isBefore(hour) ? styles.redText : null;
	}

	render(){ 
		let now  = String(moment().format('YYYY-MM-DD HH:mm:ss'));
		let expiration  = String(moment(this.props.expirationDatetime).format('YYYY-MM-DD HH:mm:ss'));

		return (
			<View>
				<View style={[styles.row, this.props.marginHorizontal && styles.marginHorizontal]}>
					{!this.props.noImage &&
						<View style={[styles.justifyContentCenter, styles.marginRight]}>
							{this.props.url ?
							<Image source={this.handleUrl(this.props.url)} style={styles.imageStyle}/> :
							<Image source={require('cotizame-native/assets/msgLogo.png')} style={styles.imageStyle}/>
							}
						</View>
					}
					<View style={styles.flex1}>
						<View style={[styles.row]}>
							<View style={styles.flex1}>
								<SmallText
									text= {this.props.label} 
									color='subdued'
								/>
								<BodyText
									text= {this.props.title}
									color='black'
									weight='medium'
								/>
							</View>
							{!this.props.noMoreButton && 
								<View>
									<TouchableOpacity>
										<Feather name='more-vertical' size={24} color={Colors.light}/>
									</TouchableOpacity>
								</View>
							}
						</View>
						<View style={[styles.row, styles.justifyContentSpaceBetween]}>
							<View style={[styles.justifyContentSpaceBetween, styles.flex1, styles.row]}>
								<View style={styles.column}>
									<SmallText
										text={Local.get('reqItem.bestPrice')}
										color='subdued'
									/>
									<SmallText 
										text={this.props.price === null ? '--' : "$ " + String(this.props.price) + " MXN"}
										color='darker'
										weight='regular'
									/>
								</View>
								<View style={styles.column}>
									<SmallText
										text={Local.get('reqItem.time')}  
										color='subdued'
									/>
									{this.state.endCountdown ?
										<SmallText
											testId="hue"
											text={Local.get('reqItem.expired')}
											color='red'
										/>
										:
										<RNMomentCountDown
						 					timeStyle={[styles.dateStyle, this.onHourLimit(expiration)]}
						 					toDate= {expiration}
						 					sourceFormatMask='YYYY-MM-DD HH:mm:ss'
						 					onCountdownEnd={this.countdownEnd}
						 				 />
									}
								</View>
							</View>
						</View>
					</View>
				</View>
			</View>
		);}
}


ReqItem.propTypes = {
	label: PropTypes.string,
	title: PropTypes.string,
	price: PropTypes.string,
	expirationDatetime: PropTypes.string,
	marginHorizontal: PropTypes.bool,
	noImage: PropTypes.bool,
	noMoreButton: PropTypes.bool,
	url: PropTypes.string,
}

ReqItem.defaultProps = {
	label: '',
	title: '',
	price: null,
	expirationDatetime: moment(0).format('YYYY-MM-DD HH:mm:ss'),
	marginHorizontal: false,
	noImage: false,
	noMoreButton: false,
	url: ''
}
