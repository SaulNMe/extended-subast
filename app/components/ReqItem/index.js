import ReqItem from  './ReqItem';

export default ReqItem;
