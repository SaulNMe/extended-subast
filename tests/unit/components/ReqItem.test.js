import React from 'react';
import { shallow } from 'enzyme';

import ReqItem from 'cotizame-native/app/components/ReqItem';
import moment from 'moment';

import {
	Image,
	TouchableOpacity
} from 'react-native';

import SmallText from 'cotizame-native/app/components/SmallText';

function createTestProps (props) {
	return {
		// common props
		label: '',
		title: '',
		price: '',
		expirationDatetime: String(moment('2137-11-24').format('YYYY-MM-DD HH:mm:ss')),
		marginHorizontal: false,
		noImage: false,
		noMoreButton: false,
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<ReqItem {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	describe('With default props', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
		it('has correct values', () => {
			let props = wrapper.instance().props;
			expect(props.expirationDatetime).toBe(moment(0).format('YYYY-MM-DD HH:mm:ss'));
		});
	});
	describe('when noImage and noMoreButton are set to True', () => {
		beforeEach(() => {
			const props = createTestProps({ noImage: true, noMoreButton: true })
			wrapper = createWrapper(props);
		});
		it('the image is not rendered', () => {
			expect(wrapper.find(Image).exists()).toBe(false);
		});
		it('the button is not rendered', () => {
			expect(wrapper.find(TouchableOpacity).exists()).toBe(false);
		});
	});
	describe('when expiration date is in the past', () => {
		beforeEach(() => {
			const props = createTestProps({ expirationDatetime: moment(0).format('YYYY-MM-DD HH:mm:ss') })
			wrapper = createWrapper(props);
		});
		it('Expired message is shown', () => {
			expect(wrapper.find(SmallText).findWhere(node=> node.prop('color') === 'red')).toBeTruthy();
		});
	});
})

// describe('callbacks');
