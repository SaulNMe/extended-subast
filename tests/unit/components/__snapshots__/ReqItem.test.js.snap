// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component>
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        false,
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "justifyContent": "center",
          },
          Object {
            "marginRight": 16,
          },
        ]
      }
    >
      <Component
        source={1}
        style={
          Object {
            "borderRadius": 10,
            "height": 60,
            "width": 60,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Component
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <Component
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <SmallText
            color="subdued"
            text=""
          />
          <BodyText
            color="black"
            text=""
            weight="medium"
          />
        </Component>
        <Component>
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <Icon
              allowFontScaling={false}
              color="#BDBDBD"
              name="more-vertical"
              size={24}
            />
          </TouchableOpacity>
        </Component>
      </Component>
      <Component
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "justifyContent": "space-between",
              },
              Object {
                "flex": 1,
              },
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Component
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <SmallText
              color="subdued"
              text="Mejor Precio"
            />
            <SmallText
              color="darker"
              text="$  MXN"
              weight="regular"
            />
          </Component>
          <Component
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <SmallText
              color="subdued"
              text="Tiempo Restante"
            />
            <_temp
              onCountdownEnd={[Function]}
              onTick={null}
              sourceFormatMask="YYYY-MM-DD HH:mm:ss"
              targetFormatMask="HH:mm:ss"
              timeStyle={
                Array [
                  Object {
                    "fontSize": 14,
                    "fontWeight": "400",
                  },
                  null,
                ]
              }
              toDate="2137-11-24 00:00:00"
            />
          </Component>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;
